src/context.cpp

```diff
@@ -78,7 +79,7 @@ NAN_METHOD(CreateContext) {
 //                         cl_int *                /* errcode_ret */) CL_API_SUFFIX__VERSION_1_0;
 NAN_METHOD(CreateContextFromType) {
   Nan::EscapableHandleScope scope;
-  REQ_ARGS(4)
+  REQ_ARGS(2) // Callbacks are not currently supported, neither user_data argument
```

Comment:
Looking to how many changes are there, I'm thinking about incremental changes to this repository, the changes need a logical break-down. Also I think it's reasonable in place of callbacks to provide native promise implementations and have async / await for these methods on the JavaScript side, such change should be carefully re-thinked since queuing callback will cause a work to be executed in the background thread which may be an unexpected behavior